### PR TITLE
Use current term in inactive transition

### LINKED ISSUE #7896: Inactive role seem to install transition steps
**Description**

When runnnig test on develop where a broker is stopped we can see in the logs that the broker transitions to inactive in the end, it looks like that the transition steps are installed

```
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 starting
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing LogStorage
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing LogStream
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing ZeebeDb
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing QueryService
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing StreamProcessor
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing SnapshotDirector
08:41:42.598 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing ExporterDirector
08:41:42.598 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 completed
```

I saw code that this is checked on each transition step, which feels a bit error prone and also confusing in regards to reading the log. I would prefer that we not install any steps on inactive step.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

The changes are more than what was intended.

1. Use current term in inactive transition. The current term is read from the context.
2. While adding new test, found out that the conditionally triggering the transitions may be error prone. I think we can rely on the PartitionTransitionSteps to optimize the transitions.
3. Updated some of the log messages. 

## Related issues

Not sure if the issue can be closed. So marking it as related

related #7896 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: broker/src/test/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionTest.java [hunk 1]
```diff
@@ -81,6 +82,48 @@ public void shouldInstallLeaderPartition() {
...
+  }
+
+  @Test
+  public void shouldUseCurrentTermForInactiveTransition() {
+    // given
+    schedulerRule.submitActor(partition);
+    when(ctx.getCurrentTerm()).thenReturn(3L);
+
+    // when -- this internally triggers transition to inactive
+    partition.onSnapshotReplicationStarted();
```
Comment (Zelldon) [inline broker/src/test/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionTest.java:120]: :x: Tbh. This looks weird. Could we just call onNewRole with inactive?

Comment (Zelldon) [reply broker/src/test/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionTest.java:120]: My thoughts: We bound our self or at least the test to the logic of this method/functionally this makes the test bridle and you have to change a test which is totally unrelated to a feature

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionTransitionProcess.java [hunk 2]
```diff
@@ -71,7 +71,9 @@ private void proceedWithTransition(final ActorFuture<Void> future) {
           startedSteps.push(nextStep);
 
           LOG.info(
-              format("Transition to %s on term %d - executing %s", role, term, nextStep.getName()));
+              format(
+                  "Transition to %s on term %d - transitioning %s",
```
Comment (Zelldon) [inline broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionTransitionProcess.java]: :+1:

**Review by ChrisKujawa — APPROVED** [review]
Thanks @deepthidevaki one smaller thing, after fixing this we can merge it. :1st_place_medal:

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartition.java [hunk 3]
```diff
@@ -292,7 +292,7 @@ private void onRoleChange(final Role newRole, final long newTerm) {
 
   private ActorFuture<Void> transitionToInactive() {
     zeebePartitionHealth.setServicesInstalled(false);
-    final var inactiveTransitionFuture = transition.toInactive();
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartition.java:295]: Raft will not change the term when transition to inactive. 
I thought about distinguishing whether inactive transition is triggered internally or triggered by raft. In first case we use the term from the context, and in the second case use the term provided via `onNewRole(role, term)`. But decided against it to make things simpler. Term is only relevant for leader transitions. So I don't think it is necessary to ensure that the term is correct.

File: broker/src/test/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionTest.java (hunk 1 above)
Comment (deepthidevaki) [reply broker/src/test/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionTest.java:120]: You are right. That was how I did it first. But I wanted to trigger an internal transition to inactive to ensure that it is using the term from the context. With `onNewRole` we are providing a term, even though for inactive transitions we are not using it.
I will change it to use `onNewRole` as you suggested.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**